sys/linux/dev_hidraw.txt

```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+include <uapi/linux/hidraw.h>
+
+resource fd_hidraw[fd]
+
+syz_open_dev$hidraw(dev ptr[in, string["/dev/hidraw#"]], id intptr, flags flags[open_flags]) fd_hidraw
+
+write$hidraw(fd fd_hidraw, data buffer[in], len bytesize[data])
+read$hidraw(fd fd_hidraw, data buffer[out], len bytesize[data])
+
+# https://elixir.bootlin.com/linux/v5.1.7/source/include/uapi/linux/hidraw.h#L34
+ioctl$HIDIOCGRDESCSIZE(fd fd_hidraw, cmd const[HIDIOCGRDESCSIZE], arg buffer[out])
+ioctl$HIDIOCGRDESC(fd fd_hidraw, cmd const[HIDIOCGRDESC], arg ptr[inout, hidraw_report_descriptor])
+ioctl$HIDIOCGRAWINFO(fd fd_hidraw, cmd const[HIDIOCGRAWINFO], arg buffer[out])
+ioctl$HIDIOCGRAWNAME(fd fd_hidraw, cmd const[HIDIOCGRAWNAME64], arg buffer[out])
+ioctl$HIDIOCGRAWPHYS(fd fd_hidraw, cmd const[HIDIOCGRAWPHYS64], arg buffer[out])
+ioctl$HIDIOCSFEATURE(fd fd_hidraw, cmd const[HIDIOCSFEATURE64], arg buffer[in])
+ioctl$HIDIOCGFEATURE(fd fd_hidraw, cmd const[HIDIOCGFEATURE64], arg ptr[inout, hidraw_get_report_arg])
+
+hidraw_report_descriptor {
+	size	int32[0:HID_MAX_DESCRIPTOR_SIZE]
+	value	array[int8, HID_MAX_DESCRIPTOR_SIZE]
+}
+
+hidraw_get_report_arg {
+	report_number	int8
+	report		array[int8]
+} [packed]
+
+define HIDIOCGRAWNAME64	HIDIOCGRAWNAME(64)
```

Comment:
If we use 64 here, should we also pass buffer of size 64? Varying just length of these arrays does trigger new interesting paths, but can cause some havoc with overlapping arguments.

Reply:
I'm not sure if it makes much difference, but I can do that. I did this the same way `dev_input.txt` does it.